third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -786,10 +786,13 @@ impl AbilitySet {
             | (Ability::Store as u8)
             | (Ability::Key as u8),
     );
+    /// Abilities for user-defined/"primitive" functions (not closures)
+    pub const DEFINED_FUNCTIONS: AbilitySet =
+        Self((Ability::Copy as u8) | (Ability::Drop as u8) | (Ability::Store as u8));
     /// The empty ability set
     pub const EMPTY: Self = Self(0);
     /// Abilities for `Functions`
-    pub const FUNCTIONS: AbilitySet = Self(Ability::Drop as u8);
+    pub const FUNCTIONS: AbilitySet = Self(0);
```

Comment:
This is for closures?

Reply:
It's the minimal set for all functions.  Defined functions have more abilities, as described above.  Closures are the only other kind of function, and they may have additional abilities but it depends on how they are constructed.  (At one point I had `drop` in this set, but realized that we may some day allow closures with captured values that can't be dropped, though I don't really know how that might work.)